src/core/lib/gprpp/arena.cc

```diff
@@ -58,7 +53,10 @@ Arena* Arena::Create(size_t initial_size) {
        GPR_CACHELINE_SIZE % GPR_MAX_ALIGNMENT == 0)
           ? GPR_CACHELINE_SIZE
           : GPR_MAX_ALIGNMENT;
-  return new (gpr_arena_malloc<alignment>(alloc_size)) Arena(initial_size);
+  void* storage = (alignment == GPR_CACHELINE_SIZE)
```

Comment:
related to the same comment above: this branch is unnecessary if you just used gpr_malloc_aligned.